### DISCUSSION TITLE
Messages from collectors in afvalwijzer?

### DISCUSSION THREAD

**Discussion by sndrsmnk** [xirixiz/homeassistant-afvalwijzer#365] [Ideas]
Hi! My municipality switches to KlikoGroep waste collector in 2025. Ofcourse they want you to download their app and 'regularly visit the website for news'. *How about no*. So i helped implement their API so i can keep using `homeassistant-afvalwijzer` with my automations and dashboard.

Aaaaanyways; i dont plan on regularly visiting their website for the latest news either and discovered there is also a `getMyMessages` API call on their front end. So now i'm wondering wether it would be nice to implement some kind of notification system in `homeassistant-afvalwijzer`?

These messages are like '*we will close collection point xyz for maintenance on dd-mm-yy*' or '*xmas tree collection, sign up here*', etc. They **could** be useful...

I would try to build this in `homeassistant-afvalwijzer` myself, but i may (will :stuck_out_tongue:) need some guidance. This would be totally new functionality and the thing is the API does not have a 'seen' flag, just a post date for the messages, so i guess i need to keep local 'seen'-state? Also i dont know if there is any other collectors offering this kind of information making this worth the effort. And not least important, wether this functionality sounds like something we (xirxiz & community) would want in this specific HACS repo.

It could as well be my first HACS repo, implementing this separated from `homeassistant-afvalwijzer`... :)

Looking forward to hearing your ideas on this.

Thanks for being awesome, everyone. :heart: Have a  great 2025. :boom:

**Comment by PatrickSt1991** [comment]
@sndrsmnk I've just created this #391, which also adds a new functionality to Afvalwijzer. Maybe we can start working on this as well.

I mean, if it's useful for you, it might be useful for other folks as well, right?

As long as we make it optional or implement it in your own garbage collector as a start, it won't interfere with the others.

**Reply by sndrsmnk** [reply]
Oops! I totally forgot about this. Just discovered the notification in my GitHub. :D

I do see there is a message in my collector's website dashboard, but it is not shown in the new `afvalwijzer notifications`-sensor yet. I'll have to pull all the recent changes and fiddle with this a bit in my dev HA. A lot has changed to the integration recently. ;)

**Comment by createthisnl** [comment]
Due to the beautiful weather here, waste collection has been postponed for the time being. 

This is clearly stated on our waste processor's website, but it is not currently being communicated via the integration. 
<img width="1095" height="145" alt="image" src="https://github.com/user-attachments/assets/6031ea04-5f33-4421-a3e6-ceef86f08e69" />
_The waste processor uses “opzet” as its platform._

After some research, I found the messages API and incorporated it into the code to create a sensor. The sensor shows how many notifications there are, and you can see the actual messages as an attribute.
<img width="445" height="771" alt="image" src="https://github.com/user-attachments/assets/d78d3d2e-1669-4530-8853-866cd386f9b5" />

This can then be displayed in your dashboard like this:
<img width="515" height="821" alt="image" src="https://github.com/user-attachments/assets/cfa77632-d7d2-423d-b2f4-e6be210f1ff5" />


Because another API needs to be called, I tried to add this as a separate, generic function. To do this, I looked at “get_waste_data_raw” and created a similar function for retrieving the messages in opzet.py.

The idea is that this can be processed in sensor_provider.py so that other platforms can also implement it.

@xirixiz I am not necessarily familiar with Python, but if there is interest, you can [see my changes](https://github.com/xirixiz/homeassistant-afvalwijzer/compare/main...createthisnl:homeassistant-afvalwijzer:2026.1002-notifications). Perhaps something useful can be done with it?

**Comment by ChefkeGremmen** [comment]
ow man, I would also really love this for mijnafvalwijzer (gemeente Oss), as our municipality also communicates these kind af message through 'news' in their app an website. Unfortunately I am not a Python developer, is there anything I can figure out?

**Reply by createthisnl** [reply]
Well, theoretically it is possible, if you mean these messages:
<img width="945" height="776" alt="image" src="https://github.com/user-attachments/assets/0f7825e4-1510-429e-8bb6-75711aaa7f15" />

However, I am not sure if the way the API is called is ideal.

The current implementation of mijnafvalwijzer only retrieves the waste data. That query does not contain any other data. 

The way I call it now, I think you get all the data from the API back. To give you an idea, that's about 800B for the waste data alone, compared to over 5MB for all the data. I can't find the parameter to retrieve only the message data. 

Currently, all data is retrieved, after which all ‘notifications’ with a ‘start_date’ of 30 days ago are displayed in the sensor. This is now filtered out because there are also old (seemingly) informative messages from 2021 and 2022 that do not really change.

Perhaps @xirixiz knows more about this?

**Reply by createthisnl** [reply]
This implementation has been added in https://github.com/xirixiz/homeassistant-afvalwijzer/pull/485
